mllib/src/main/scala/org/apache/spark/mllib/stat/test/KolmogorovSmirnovTest.scala

```diff
@@ -190,5 +191,93 @@ private[stat] object KolmogorovSmirnovTest extends Logging {
     val pval = 1 - new KolmogorovSmirnovTest().cdf(ksStat, n.toInt)
     new KolmogorovSmirnovTestResult(pval, ksStat, NullHypothesis.OneSampleTwoSided.toString)
   }
+
+  /**
+   * Implements a two-sample, two-sided Kolmogorov-Smirnov test, which tests if the 2 samples
+   * come from the same distribution
+   * @param data1 `RDD[Double]` first sample of data
+   * @param data2 `RDD[Double]` second sample of data
+   * @return [[org.apache.spark.mllib.stat.test.KolmogorovSmirnovTestResult]] with the test
+   *        statistic, p-value, and appropriate null hypothesis
+   */
+  def testTwoSamples(data1: RDD[Double], data2: RDD[Double]): KolmogorovSmirnovTestResult = {
+    val n1 = data1.count().toDouble
+    val n2 = data2.count().toDouble
+    val isSample1 = true // identifier for sample 1, needed after co-sort
+    // combine identified samples
+    val joinedData = data1.map(x => (x, isSample1)) ++ data2.map(x => (x, !isSample1))
```

Comment:
Would it be clearer to write `true` and `false` instead of `isSample1` and `!isSample1`? I don't have a strong opinion.
Can the `map` functions be like `.map((_, isSample1))` for tidiness or does that syntax not work here?
Finally I wonder if `.union` is clearer than `++` here? I don't have a strong opinion either, just am somehow used to method invocations on RDDs and `++`-like syntax for Scala collections.